src/Components/test/testassets/BasicTestApp/EventCustomArgsTypes.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Microsoft.AspNetCore.Components;
+
+namespace BasicTestApp.CustomEventTypesNamespace
+{
+    [EventHandler("ontestevent", typeof(TestEventArgs), true, true)]
+    [EventHandler("onkeydown.testvariant", typeof(TestKeyDownEventArgs), true, true)]
+    [EventHandler("onkeydown.yetanother", typeof(YetAnotherCustomKeyboardEventArgs), true, true)]
+    [EventHandler("oncustommouseover", typeof(EventArgs), true, true)]
+    public static class EventHandlers
```

Comment:
I'm ok if we think this is out of the scope of this PR, but I truly think we should change the way this works to avoid having to use this class moving forward.

Reply:
Understood.

I agree we could make this nicer. But `[EventHandler]` is what we've already got and need to remain back-compatible with. This PR doesn't change the .NET-side registration mechanism. If we want to create a different and nicer registration mechanism, it can be looked at separately if customer demand justifies it. This PR doesn't lock us any more closely to the existing registration style.